### ISSUE TITLE
Incompatiblity with SSG

### ISSUE THREAD

**Issue by MrSunshyne** [harlan-zw/nuxt-og-image#35]
### Details

After almost a month of debugging(my personal website), I finally found in [the readme ](https://github.com/harlan-zw/nuxt-og-image#ogimagedynamic--defineogimagedynamic) today: 

`This feature is not compatible with static sites built using nuxi generate.`
 
While reading more I can see the `forcePrerender` option which says 
`This is enabled when you run nuxi generate. It forces all OG images to be pre-rendered as the server is not available to generate runtime images.`

These seem contradictory to me. But perhaps i misunderstood something. Is it possible to use `nuxt-og-image` with SSG at all or should I avoid it completely? I love what this module does and would like to use it but my website is SSG.

For info I experienced a nasty behavior due to this which [led](https://github.com/nuxt/nuxt/issues/13369#issuecomment-1493287051) me [down this](https://discord.com/channels/473401852243869706/1078259283579387954/1081903841148162069) [rabbit](https://github.com/MrSunshyne/nuxt-debug-usefetch) [hole](https://github.com/MrSunshyne/nuxt3-ssg-bug-reproduction). After deploying my site to a static hosting, api calls were still being made on the client side even thought the full site had been build with `nuxi generate`. Since my api server was on localhost, the site broke on prod and it only today that i was able to narrow it down to this module.

**Comment by harlan-zw** [comment]
Hey @MrSunshyne, thanks for the issue.

Yes, SSG should be the safest option for using the module. These issues don't really seem related though.

Do you have any specific errors you could provide related to the module?

**Comment by harlan-zw** [comment]
I've added some SSG specific details to the docs https://github.com/harlan-zw/nuxt-og-image#ssg-requirements

**Comment by MrSunshyne** [comment]
hey @harlan-zw thanks for the reply. I was able to create a reproduction of the issue. I [created a repo with Nuxt 3](https://github.com/MrSunshyne/nuxt-debug-usefetch), and it's a simple blog-like site with the following structure: 

```md
- /  (lists all blog posts)
- /blog  (lists all blog posts)
- /blog/<slug> (shows one individual blog post and contains one <OgImageStatic /> component)
 ```

The site uses a simple server route `/api/posts` to retrieve a list of posts via `useFetch`

I deployed on version without `nuxt-og-image` and one version with.

without nuxt-og-image : https://nuxt-debug-usefetch.vercel.app/
with nuxt-og-image : https://with-nuxt-image.vercel.app/

Both are deployed using `nuxi generate`

## What's the bug?
When I use nuxt-og-image, after I generate and deploy the static site, the route where I list all blog posts makes an api call to `/api/posts` **on the client side.** Because that route doesn't actually exist on a static site, it shows a blank list. I think this should not be hapenning.

## What's the expected behavior
useFetch should not be doing any api call on the client side since this is a static site.

I tried figuring out why this behaviour happens, but i wasn't able to come up with anything so far. Perhaps you will have a better idea. Thanks again

**Comment by harlan-zw** [comment]
Thanks for the extra detail and reproduction. I could replicate it with your setup.

Doing some testing, the issue seems to be fixed by using the latest beta `"nuxt-og-image": "2.0.0-beta.13"`, could you confirm?

**Comment by MrSunshyne** [comment]
happy to report that the update version FIXED the issue. I tested it here: https://nuxt-og-2-beta.vercel.app/ [branch](https://github.com/MrSunshyne/nuxt-debug-usefetch/tree/nuxt-og-2-beta)

The only change is the package version. Thank you so much for this awesome work 🎉

If it's possible to get an overview from you, I would love to learn what the issue was and how the new version addressed it.

**Comment by harlan-zw** [comment]
I would love to help but I honestly have no idea. You could look through the v2 changelog so far but I've worked on many improvements so would be difficult to pinpoint.

v2 will be released stable soon hopefully

**Comment by MrSunshyne** [comment]
@harlan-zw hey that might have been a cache of `main` we were looking at... It seems the issue still exists :( 

https://nuxt-og-2-beta.vercel.app/

so v2 didn't address this particular issue 😢

**Comment by harlan-zw** [comment]
Okay, I think I have found an issue. For whatever reason when prepending a URL to the `x-nitro-prerender`, before the `payload` prerender, it breaks the payload prerender.

I'm guessing this is related to some passed context within the local fetch that is lost. This can probably be fixed in Nuxt by prioritizing the payload render to be first. I'll need to do some more testing and reproductions around specifically this.

For now, I've just moved the logic around so the payload will always be prerendered first. You can try v2.0.0-beta.14

**Comment by harlan-zw** [comment]
Hey @MrSunshyne 

Just to give you an update, I believe I've found the issue in the Nuxt core and made a PR to fix it. https://github.com/nuxt/nuxt/pull/20590

Hopefully we can get this closed off soon now

**Comment by harlan-zw** [comment]
Looks to be working in the latest nuxt edge :) The bug fix will land in Nuxt 3.5.0

```
    "nuxt": "npm:nuxt3@3.5.0-28047441.d2170b80",
    "nuxt-og-image": "2.0.0-beta.28"
```

Will close for now but let me know if you noticee anything weird

**Comment by MrSunshyne** [comment]
@harlan-zw I tested it with your fix on nuxt3.5-x, can confirm it is working properly now ! I'm [using it on my blog](https://sandeep.ramgolam.com/blog/developers-institute-x-front-end-coders-april-meetup-2023) and it doesn't break SSG anymore. Thank you so much for locating and fixing this.
